module-apps/apps-common/widgets/DigitsContainer.hpp

```diff
@@ -16,47 +16,19 @@ namespace gui
         std::uint32_t minusWidth;
         std::uint32_t leftBoxWidth;
         std::uint32_t rightBoxWidth;
+        std::uint32_t avaliableEvenWidth;
+        bool isNotCentered{false};
     };
 
-    template <size_t N> struct DigitsContainer
+    struct DigitsContainer
     {
-        std::array<Label *, N> digits;
+        explicit DigitsContainer(std::uint32_t digitsCount);
+        std::vector<Label *> digits;
+
         void setMinutesBox(std::uint32_t minutes, DimensionsParams params);
         void setSecondsBox(std::uint32_t seconds, DimensionsParams params);
     };
-    template <size_t N> void setDigits(std::string text, const DigitsContainer<N> &container, DimensionsParams params)
-    {
-        std::for_each(std::crbegin(container.digits),
-                      std::crend(container.digits),
-                      [text = std::move(text), &params](Label *label) mutable {
-                          if (text.empty()) {
-                              label->setText("");
-                              label->setSize(0, 0);
-                          }
-                          else {
-                              label->setText(std::string{text.back()});
-                              label->setSize(params.digitMaxWidth, params.mainBoxHeight);
-                              text.pop_back();
-                          }
-                      });
-    }
-
-    template <size_t N> void DigitsContainer<N>::setMinutesBox(std::uint32_t minutes, DimensionsParams params)
-    {
-        auto rangeGuard = 1000;
-        minutes %= rangeGuard;
-
-        setDigits(std::to_string(minutes), *this, params);
-    }
-
-    template <size_t N> void DigitsContainer<N>::setSecondsBox(std::uint32_t seconds, DimensionsParams params)
-    {
-        auto rangeGuard = 100;
-        seconds %= rangeGuard;
 
-        bool shouldBeDisplayedAsOneDigit = seconds < 10;
-        std::string text = shouldBeDisplayedAsOneDigit ? "0" + std::to_string(seconds) : std::to_string(seconds);
+    void setDigits(std::string text, const DigitsContainer &container, DimensionsParams params);
```

Comment:
Move it to cpp file? As far as I know, it is not used anywhere else?

Reply:
Passing by value should be okay thanks to the SSO(short strings optimization).

Reply:
Its declaration in hpp definition is in cpp, so I can remove definicion in hpp file

Reply:
Its declaration in hpp definition is in cpp, so I can remove definicion in hpp file asd

Reply:
Yeah, if it is not used by another module, there is no need to have it in public API.